e2e/test/scenarios/embedding/embedding-full-app.cy.spec.js

```diff
@@ -315,6 +326,64 @@ describeEE("scenarios > embedding > full app", () => {
 
       cy.findByTestId("main-logo").should("be.visible");
     });
+
+    it("should have parameters header occupied the entire horizontal space when visiting a dashboard via navigation (metabase#30645)", () => {
+      const dashboardDetails = {
+        name: "interactive dashboard embedding",
+        parameters: [
+          {
+            id: "50c9eac6",
+            name: "ID",
+            slug: "id",
+            type: "id",
+          },
+        ],
+      };
+      cy.createDashboard(dashboardDetails).then(
+        ({ body: { id: dashboardId } }) => {
+          const card = createTextCard({
+            col: 0,
+            row: 0,
+            size_x: 6,
+            size_y: 20,
+            text: "I am a very long text card",
+          });
+          updateDashboardCards({ dashboard_id: dashboardId, cards: [card] });
+        },
+      );
+
+      visitFullAppEmbeddingUrl({ url: "/" });
+
+      cy.log("Navigate to a dashboard via in-app navigation");
+      navigationSidebar().findByText("Our analytics").click();
+      cy.findByRole("main").findByText(dashboardDetails.name).click();
+      closeNavigationSidebar();
+      navigationSidebar().findByText("Our analytics").should("not.be.visible");
+
+      cy.get("main header")
+        .findByText(dashboardDetails.name)
+        .should("be.visible");
+      getDashboardCard()
+        .findByText("I am a very long text card")
+        .should("be.visible");
+
+      // The bug won't appear if we scroll instantly and check the position of the dashboard parameter header.
+      // I suspect that happens because we used to calculate the dashboard parameter header position in JavaScript,
+      // which could take some time.
+      const FPS = 1000 / 60;
+      cy.findByRole("main").scrollTo("bottom", { duration: 2 * FPS });
```

Comment:
I put this here, though it won't be apparent why this is necessary. But it's necessary for reproducing the issue. Without `duration` option, I couldn't fail the test on `0.46.1`.